src/lerobot/datasets/dataset_tools.py

```diff
@@ -0,0 +1,998 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dataset tools utilities for LeRobotDataset.
+
+This module provides utilities for:
+- Deleting episodes from datasets
+- Splitting datasets into multiple smaller datasets
+- Adding/removing features from datasets
+- Merging datasets (wrapper around aggregate functionality)
+"""
+
+import logging
+import shutil
+from collections.abc import Callable
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import torch
+from tqdm import tqdm
+
+from lerobot.datasets.aggregate import aggregate_datasets
+from lerobot.datasets.compute_stats import aggregate_stats
+from lerobot.datasets.lerobot_dataset import LeRobotDataset, LeRobotDatasetMetadata
+from lerobot.datasets.utils import (
+    DEFAULT_CHUNK_SIZE,
+    DEFAULT_DATA_FILE_SIZE_IN_MB,
+    DEFAULT_DATA_PATH,
+    DEFAULT_EPISODES_PATH,
+    get_parquet_file_size_in_mb,
+    to_parquet_with_hf_images,
+    update_chunk_file_indices,
+    write_info,
+    write_stats,
+    write_tasks,
+)
+from lerobot.utils.constants import HF_LEROBOT_HOME
+
+
+def _load_episode_with_stats(src_dataset: LeRobotDataset, episode_idx: int) -> dict:
+    """Load a single episode's metadata including stats from parquet file.
+
+    Args:
+        src_dataset: Source dataset
+        episode_idx: Episode index to load
+
+    Returns:
+        dict containing episode metadata and stats
+    """
+    ep_meta = src_dataset.meta.episodes[episode_idx]
+    chunk_idx = ep_meta["meta/episodes/chunk_index"]
+    file_idx = ep_meta["meta/episodes/file_index"]
+
+    parquet_path = src_dataset.root / DEFAULT_EPISODES_PATH.format(chunk_index=chunk_idx, file_index=file_idx)
+    df = pd.read_parquet(parquet_path)
+
+    episode_row = df[df["episode_index"] == episode_idx].iloc[0]
+
+    return episode_row.to_dict()
+
+
+def delete_episodes(
+    dataset: LeRobotDataset,
+    episode_indices: list[int],
+    output_dir: str | Path | None = None,
+    repo_id: str | None = None,
+) -> LeRobotDataset:
+    """Delete episodes from a LeRobotDataset and create a new dataset.
+
+    Args:
+        dataset: The source LeRobotDataset.
+        episode_indices: List of episode indices to delete.
+        output_dir: Directory to save the new dataset. If None, uses default location.
+        repo_id: Repository ID for the new dataset. If None, appends "_filtered" to original.
+    """
+    if not episode_indices:
+        raise ValueError("No episodes to delete")
+
+    valid_indices = set(range(dataset.meta.total_episodes))
+    invalid = set(episode_indices) - valid_indices
+    if invalid:
+        raise ValueError(f"Invalid episode indices: {invalid}")
+
+    logging.info(f"Deleting {len(episode_indices)} episodes from dataset")
+
+    if repo_id is None:
+        repo_id = dataset.repo_id
```

Comment:
If `output_dir` and `repo_id` are both `None`, the script will write the new dataset to the same location as the original, effectively overwriting it. This could lead to accidental data loss.

```suggestion
        repo_id = f"{dataset.repo_id}_filtered"
```